src/components/Layout/mdx/If.tsx

```diff
@@ -1,20 +1,36 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import { useLayoutContext } from 'src/contexts/layout-context';
 import { LanguageKey } from 'src/data/languages/types';
+import UserContext from 'src/contexts/user-context';
 
 interface IfProps {
-  lang: LanguageKey;
+  lang?: LanguageKey;
+  loggedIn?: boolean;
   className?: string;
   children: React.ReactNode;
   as?: React.ElementType;
 }
 
-const If: React.FC<IfProps> = ({ lang, children }) => {
+const If: React.FC<IfProps> = ({ lang, loggedIn, children }) => {
   const { activePage } = useLayoutContext();
   const { language } = activePage;
-  const splitLang = lang.split(',');
+  const userContext = useContext(UserContext);
 
-  return splitLang.includes(language) ? children : null;
+  let shouldShow = true;
+
+  // Check language condition if lang prop is provided
+  if (lang !== undefined && language) {
+    const splitLang = lang.split(',');
+    shouldShow = shouldShow && splitLang.includes(language);
+  }
+
+  // Check logged in condition if loggedIn prop is provided
+  if (loggedIn !== undefined) {
```

Comment:
I would check for userContext.sessionState as well here, we may not always have that and don't want the access on the following line to error